docs/modules/ROOT/pages/explanations/connectiondetails.adoc

```diff
@@ -0,0 +1,196 @@
+With composition functions the `ConnectionDetailTypeFromConnectionSecretKey` is probably most commonly used, as field values can easily be accessed and set as an `xfnv1alpha1.ExplicitConnectionDetail` in the composite.
+
+[source,golang]
+----
+cd := []xfnv1alpha1.DerivedConnectionDetail{
+	{
+		Name:                    pointer.String(secretKeyName),
+		FromConnectionSecretKey: pointer.String(secretKeyName),
+		Type:                    xfnv1alpha1.ConnectionDetailTypeFromConnectionSecretKey,
+	},
+	{
+		Name:                    pointer.String(accessKeyName),
+		FromConnectionSecretKey: pointer.String(accessKeyName),
+		Type:                    xfnv1alpha1.ConnectionDetailTypeFromConnectionSecretKey,
+	},
+}
+
+return iof.Desired.PutWithResourceName(ctx, user, "minio-user", runtime.AddDerivedConnectionDetails(cd))
+----
+
+Lastly, the `User` xrd needs to be made aware of the keys it should expose in the claim connection details.
+This is done in `component-appcat`.
+
+[source,jsonnet]
+----
+local connectionSecretKeys = [
+  'AWS_ACCESS_KEY_ID',
+  'AWS_SECRET_ACCESS_KEY',
+];
+
+local xrd = xrds.XRDFromCRD(
+  ...
+  connectionSecretKeys=connectionSecretKeys,
+);
+----
+
+=== Helm example
+
+These connection details differ a bit from normal managed resources, as the provider cannot now what secrets every available helm chart can expose.
+
+Instead, provider-helm exposes a flexible way to specify exactly which of the helm managed object contain the relevant values.
+
+Let's assume our given helm chart writes a secret and a service.
+Both of them contain information relevant to the user to access the service.
+
+[source,golang]
+----
+r := &v1beta1.Release{
+		Spec: v1beta1.ReleaseSpec{
+			ResourceSpec: xpv1.ResourceSpec{
+				WriteConnectionSecretToReference: &xpv1.SecretReference{ <1>
+					Name: "myname",
+					Namespace: "secretnamespace",
+				},
+			},
+			ConnectionDetails: []v1beta1.ConnectionDetail{
+				{
+					ObjectReference: corev1.ObjectReference{
+						Kind:      "Secret",
+						Name:      "mysecret",
+						Namespace: "myns",
+						FieldPath: "data.my-password", <2>
+					},
+					ToConnectionSecretKey: "connectionKey",
+				},
+				{
+					ObjectReference: corev1.ObjectReference{
+						Kind:      "Service",
+						Name:      "myservice",
+						Namespace: "myns",
+						FieldPath: "spec.ports[0].port",
+					},
+				},
+			},
+		},
+	}
+----
+<1> The connection detail still needs to written somewhere
+<2> We don't have to bother with base64 encoding, provider-helm will handle this for us
+
+With this method it should be easier to handle most cases.
+However, provider-helm's connection detail management isn't perfect.
+It cannot handle any transformations on the secrets.
+So if anything needs to concatenated, split, or changed in any other way, then this might unfortunately not be the way to do it.
+
```

Comment:
It should probably also be documented that you need to add that connection details to the composite as pointed in the above example with `return iof.Desired.PutWithResourceName(ctx, user, "minio-user", runtime.AddDerivedConnectionDetails(cd))`